Appl 8689 merge updates from base repo

jonobarel-slatescience opened the pull request:
Updated branch with all changes from upstream gree/unity-webview.

KojiNakamaru committed:
modified to support onHttpError on osx. (#883)

KojiNakamaru committed:
updated binaries.

FumiyaHr committed:
Add notification of window close in separated mode (#886)

KojiNakamaru committed:
updated binaries.

KojiNakamaru committed:
fixed to add queries for camera in AndroidManifest.xml. (#891)

KojiNakamaru committed:
updated binaries.

KojiNakamaru committed:
updated README.md.

KojiNakamaru committed:
fixed GetCookies to work with recent iOS/OSX and to invoke the 'cookes:' callback. (#895)

KojiNakamaru committed:
updated binaries.

KojiNakamaru committed:
fixed call for GetCookies (cf. #895).

KojiNakamaru committed:
fixed to utilize UnityEngine.Android.Permissions instead of directly request permissions in java (cf. #890). (#901)

KojiNakamaru committed:
fixed SampleWebView.cs to check whether webViewObject exists. (#902)

KojiNakamaru committed:
updated binaries.

KojiNakamaru committed:
fixed RequestFileChooserPermissions to support 2020.1 for former versions (cf. https://github.com/gree/unity-webview/commit/38493dc5893236ac4a3e44325b44f5ba738c6ea5#r106960472 ). (#903)

KojiNakamaru committed:
updated binaries.

KojiNakamaru committed:
[experimental] introduced saveDataURL for iOS. (#904)

KojiNakamaru committed:
updated binaries.

KojiNakamaru committed:
merged 896ebe5b89e633ba82561bbe906b614d146f6a46 into plugins/iOS/WebViewWithUIWebView.mm. (#906)

KojiNakamaru committed:
modified to inject Unity.call definition by WKUserScript. (#907)

KojiNakamaru committed:
modified sample app to test saveDataURL and to save html contents under temporaryCachePath insteadOf persistentDataPath. (#908)

KojiNakamaru committed:
updated binaries.

KojiNakamaru committed:
fixed SampleWebView.cs about the variable "js".

KojiNakamaru committed:
modified to add GET_CONTENT in queries in AndroidManifest.xml. (#911)

KojiNakamaru committed:
updated binaries.

KojiNakamaru committed:
fixed missing AddGallery() for UNITY_2018_1_OR_NEWER. (#912)

KojiNakamaru committed:
updated binaries.

KojiNakamaru committed:
fixed memory leaks of script message handlers. (#917)

KojiNakamaru committed:
updated binaries.

KojiNakamaru committed:
fixed setMargins in unity-webview.js (not for 2020 or later). (#920)

KojiNakamaru committed:
updated binaries.

KojiNakamaru committed:
fixed iOS code to set inspectable for iOS 16.4+ and development build. (#925)

KojiNakamaru committed:
updated binaries.

KojiNakamaru committed:
fixed android keyboard detection. (#928)

KojiNakamaru committed:
updated binaries.

KojiNakamaru committed:
updated sample/.gitignore.

KojiNakamaru committed:
delayed to call OnRequestFileChooserPermissionsResult (cf. https://github.com/gree/unity-webview/issues/890#issuecomment-1600345675 ). (#937)

KojiNakamaru committed:
introduced UNITYWEBVIEW_ANDROID_FORCE_MARGIN_ADJUSTMENT_FOR_KEYBOARD. (#938)

KojiNakamaru committed:
updated binaries.

KojiNakamaru committed:
disabled android visibility control on pause/resume. (#945)

KojiNakamaru committed:
updated binaries.

KojiNakamaru committed:
fixed to modify mCustomHeaders in the UI thread (cf. #951 ). (#952)

KojiNakamaru committed:
adjusted Rakefile about UNITY path.

KojiNakamaru committed:
updated binaries.

KojiNakamaru committed:
modified to define pseudo requestAnimationFrame by WKUserScript. (#953)

KojiNakamaru committed:
updated binaries.

KojiNakamaru committed:
introduced Pasue/Resume for iOS (only for media). (#955)

KojiNakamaru committed:
updated binaries.

KojiNakamaru committed:
fixed to explicitly invoke requestLayout() for uniy issue 1361532. (#983)

KojiNakamaru committed:
updated binaries.

KojiNakamaru committed:
fixed to bring layout to front for UNITY_2022_1_OR_NEWER. (#985)

KojiNakamaru committed:
updated binaries.

trapezoid committed:
allow 'about:srcdoc' URL in iOS WKWebView for iframe's srcdoc attribute (#988)

Co-authored-by: haruto.otake <haruto.otake@dena.jp>

KojiNakamaru committed:
applied b9f756c962892added7182017d9eb15f8ff8577e to plugins/iOS/WebViewWithUIWebView.mm. (#990)

KojiNakamaru committed:
updated binaries.

KojiNakamaru committed:
modified to support android api level 33. (#993)

KojiNakamaru committed:
updated binaries.

KojiNakamaru committed:
fixed to set mWebView = null in ui thread (mWebView should always be modifed in UI thread). (#997)

KojiNakamaru committed:
updated binaries.

KojiNakamaru committed:
fixed android Update() to do nothing when the app is paused. (#998)

KojiNakamaru committed:
fixed to enable forceBringToFront also for 2021.x. (#999)

KojiNakamaru committed:
updated binaries.

KojiNakamaru committed:
modified SetVisibility to invoke layout.bringToFront() if forceBringToFront. (#1000)

KojiNakamaru committed:
updated binaries.

KojiNakamaru committed:
fixed to retrieve the window visible frame height at start and to update the current height for the current orientation. (#1006)

KojiNakamaru committed:
updated binaries.

KojiNakamaru committed:
fixed to compute keyboardHeight from DrawingRect and WindowVisibleDisplayFrame. (#1008)

KojiNakamaru committed:
updated binaries

KojiNakamaru committed:
fixed to set android:windowSoftInputMode to adjustPan. (#1009)

KojiNakamaru committed:
updated binaries.

KojiNakamaru committed:
fixed to use using and Dispose() for every AndroidJavaObject (cf. https://github.com/gree/unity-webview/issues/1012 ). (#1013)

KojiNakamaru committed:
updated binaries.

KojiNakamaru committed:
modified to refocus activeElement for forcibly making it visible on rotation. (#1014)

KojiNakamaru committed:
fixed #1014 . (#1015)

KojiNakamaru committed:
updated binaries.

KojiNakamaru committed:
modified to build release/development WebViewPlugin.aar and to pick one of them for EditorUserBuildSettings.development (#1017)

* modified to build release aar and development aar.

* removed obsolete files.

* updated Rakefile to utilize new install.sh.

* updated UnityWebViewPostprocessBuild.cs to emit WebViewPlugin.aar for EditorUserBuildSettings.development.

* fixed cleanup logic before build.

* updated binaries.

KojiNakamaru committed:
updated plugins/Android/install.sh for simplifying the setup. (#1019)

* adjusted install.sh for android for simplying the setup.

* updated install.sh to support git bash on windows.

KojiNakamaru committed:
updated README.md about Git Bash.

KojiNakamaru committed:
fixed to temporarily generate Assets/Plugins/Android/WebViewPlugin.aar instead of Temp/StagingArea/aar/WebViewPlugin.aar. (#1028)

KojiNakamaru committed:
updated binaries.

hanachiru committed:
Fixed a bug in relative margins on Android

KojiNakamaru committed:
Merge pull request #1035 from hanachiru/feature/android_relative_margin

Fixed a bug in relative margins on Android

KojiNakamaru committed:
updated binaries.

KojiNakamaru committed:
fixed to get correct iw and ih for non full screen mode.

KojiNakamaru committed:
Merge pull request #1036 from gree/fix/android_iw_ih_for_non_fullscreen_case

fixed to get correct iw and ih for non full screen mode.

KojiNakamaru committed:
fixed UnityWebViewPostprocessBuild.cs to perform File.Copy(..., true).

KojiNakamaru committed:
Merge pull request #1037 from gree/fix/file_copy_with_overwrite

fixed UnityWebViewPostprocessBuild.cs to perform File.Copy(..., true).

KojiNakamaru committed:
updated binaries.

KojiNakamaru committed:
implemented shouldInterceptRequest for API level 21 or later.

KojiNakamaru committed:
Merge pull request #1046 from gree/fix/android_should_intercept_request

implemented shouldInterceptRequest for API level 21 or later.

KojiNakamaru committed:
fixed typo in #1046 .

KojiNakamaru committed:
Merge pull request #1047 from gree/fix/typo_in_1046

fixed typo in #1046 .

KojiNakamaru committed:
updated binaries.

KojiNakamaru committed:
adjusted SetVisibility(false) to invoke document.activeElement.blur() for closing the keyboard.

KojiNakamaru committed:
Merge pull request #1051 from gree/feature/blur_on_visibility_false

adjusted SetVisibility(false) to invoke document.activeElement.blur() for closing the keyboard.

KojiNakamaru committed:
updated binaries.

Harrison-Dev committed:
fix "gamma correction" issue on linear space

KojiNakamaru committed:
Merge pull request #1052 from Harrison-Dev/fix/gamma_correction

fix "gamma correction" issue on linear space

KojiNakamaru committed:
updated binaries.

KojiNakamaru committed:
fixed #1055 .

KojiNakamaru committed:
Merge pull request #1056 from gree/fix/osx_update_bug

fixed #1055 (inRendering should be turned on/off in if (refreshBitmap) clause).

KojiNakamaru committed:
updated binaries.

KojiNakamaru committed:
modified plugins/Mac/Sources/WebView.mm to utilize an invisible window for the non-separated mode.

KojiNakamaru committed:
Merge pull request #1061 from gree/feature/osx_impl_refined

modified plugins/Mac/Sources/WebView.mm to utilize an invisible window for the non-separated mode.

KojiNakamaru committed:
updated binaries.

KojiNakamaru committed:
introduced WebViewObject.canvas for osx for instanciating a background image that prevent events pass through.

KojiNakamaru committed:
Merge pull request #1062 from gree/feature/osx_ugui_bg

introduced WebViewObject.canvas for osx for instanciating a background image that prevent events pass through.

KojiNakamaru committed:
fixed #1062 about OnDestroy and SetVisibility.

KojiNakamaru committed:
Merge pull request #1063 from gree/fix/1062

fixed #1062 about OnDestroy and SetVisibility.

KojiNakamaru committed:
updated binaries.

KojiNakamaru committed:
fixed to enclose bg-related code with "#if UNITY_EDITOR_OSX || UNITY_STANDALONE_OSX".

KojiNakamaru committed:
Merge pull request #1064 from gree/fix/osx_bg

fixed to enclose bg-related code with "#if UNITY_EDITOR_OSX || UNITY_STANDALONE_OSX".

KojiNakamaru committed:
updated binaries.

KojiNakamaru committed:
fixed to hide a window in non-separated mode (cf. #1066).

KojiNakamaru committed:
Merge pull request #1067 from gree/fix/osx_visible_window

fixed to hide a window in non-separated mode.

KojiNakamaru committed:
updated binaries.

KojiNakamaru committed:
introduced devicePixelRatio for osx.

KojiNakamaru committed:
Merge pull request #1069 from gree/feature/osx_device_pixel_ratio

introduced devicePixelRatio for osx.

KojiNakamaru committed:
updated binaries.

KojiNakamaru committed:
fixed AdjustBottomMargin() for "windowed" mode with "status bar" and "render outside safe area".

KojiNakamaru committed:
Merge pull request #1077 from gree/fix/adjustbottommargin

fixed AdjustBottomMargin() for "windowed" mode with "status bar" and "render outside safe area".

KojiNakamaru committed:
updated binaries.

KojiNakamaru committed:
fixed plugins/Android/install.sh about incorrectly swapped release/development builds.

KojiNakamaru committed:
Merge pull request #1081 from gree/fix/android_release_development

fixed plugins/Android/install.sh about incorrectly swapped release/development builds.

KojiNakamaru committed:
updated binaries.

KojiNakamaru committed:
modified install.sh to remove the zorder patch for 5.6.x by default.

KojiNakamaru committed:
Merge pull request #1082 from gree/feature/install_sh_zorderpatch_option

modified install.sh to remove the zorder patch for 5.6.x by default.

KojiNakamaru committed:
updated binaries.

KojiNakamaru committed:
introduced SetMixedContentMode (cf. #1084).

KojiNakamaru committed:
added a comment about SetMixedContentMode in SampleWebView.cs.

KojiNakamaru committed:
Merge pull request #1085 from gree/feature/setmixedcontentmode

introduced SetMixedContentMode

KojiNakamaru committed:
updated binaries.

jonobarel-slatescience committed:
APPL-8689 Updated and merged changes from base repo

jonobarel-slatescience committed:
AAPL-8689 testing new version of the compiled plugin for android

jonobarel-slatescience committed:
AAPL-8689 updated aar file after recompile

jonobarel-slatescience committed:
Modifiy Install process to not erase CUnity* files.

jonobarel-slatescience committed:
modifying .gitignore to include compiled plugins folder  AAPL-8689

jonobarel-slatescience committed:
Recompiled plugins with Matific-specific changes AAPL-8689

jonobarel-slatescience committed:
added comment AAPL-8689

lavikabel commented on build/.gitignore:
```diff
@@ -1,2 +1,6 @@
 Packager/Assets/Plugins
 Packager/Library
+
+#MATIFIC SPECIFIC
```
why?

jonobarel-slatescience replied:
I want the plugin files to be included in the repo.

lavikabel replied:
hmmm why?

jonobarel-slatescience replied:
no longer relevant.

lavikabel commented on build/Rakefile:
```diff
@@ -20,7 +20,7 @@
 require 'fileutils'
 
-UNITY="/Applications/UnityCurrent/Unity.app/Contents"
+UNITY="/Applications/Unity/Hub/Editor/2019.4.40f1/Unity.app/Contents/MacOS/Unity"
```
Are the changes here yours?

jonobarel-slatescience replied:
No. These are from the main repo.

lavikabel commented on plugins/Android/MAD_README.md:
```diff
@@ -12,7 +12,8 @@ This is a custom unity webview fork to allow pausing and resuming unity while th
  	 
 1. `plugins/Android/local.properties` needs to be configured w/ path to Android SDK used by Unity (i.e. `sdk.dir=/Applications/Unity/Hub/Editor/2020.3.30f1/PlaybackEngines/AndroidPlayer/SDK`)
 2. Goto `plugins\Android` and open `install.sh`. (**Don't use `install-nofragment.sh`**)
```
the nofragment parts are no longer needed, right?

jonobarel-slatescience replied:
The parameter still exists. I should add a note about --development, though.

lavikabel commented on plugins/Android/install.sh:
```diff
@@ -1,51 +1,128 @@
-    '--mono' )
-        SCRIPTING_BACKEND="mono"
+    '--zorderpatch')
+        UNITY='2022.3.32f1'
```
we dont need the `zorderpatch` flag for us it is always the same unity

jonobarel-slatescience replied:
I didn't touch this, only the unity version.

lavikabel replied:
ok, revert to what gree had, one less conflict in the future.
We are not passing this flag anyway

jonobarel-slatescience committed:
Updated MAD_README.md with instructions for the current version of Gree/Unity-Webview

-- support for --development flag when compiling
-- including the generated .tmpl files in the repo

jonobarel-slatescience committed:
including android compiled plugins AAPL-8689

jonobarel-slatescience committed:
removing plugins from .gitignore in order to update compiled libraries in repo AAPL-8689

jonobarel-slatescience committed:
remove .gitignore

jonobarel-slatescience committed:
Removed plugin files.

jonobarel-slatescience committed:
Imported newly built plugin files AAPL-8689

jonobarel-slatescience committed:
updated plugins in CORRECT directory APPL-8689

lavikabel commented on build/.gitignore:
```diff
@@ -1,6 +0,0 @@
-Packager/Assets/Plugins
-Packager/Library
```
why did you remove this file?

jonobarel-slatescience replied:
Restored the file. Thought it was blocking the tracking of the .tmpl files but something else was doing it.

lavikabel commented on plugins/Android/MAD_README.md:
```diff
@@ -12,31 +12,46 @@ This is a custom unity webview fork to allow pausing and resuming unity while th
  
 
 ## Deploying new Plugin in git
-1. Copy `WebViewPlugin.aar` and `core-1.6.0.aar.tmpl` from `build/Packager/Assets/Plugins/Android` to `dist/package/Assets/Plugins/Android`
-2. Push to git `WebViewPlugin.aar` and `core-1.6.0.aar.tmpl` from previous section.
```
Why did you remove those steps?

jonobarel-slatescience replied:
See updated version, the instructions are updated to include these again as I thought the project structure had changed.

lavikabel commented on plugins/Android/MAD_README.md:
```diff
@@ -12,31 +12,46 @@ This is a custom unity webview fork to allow pausing and resuming unity while th
+3. In your SlateApps's branch - edit `Packages/manifest.json` and replace the current hash with that of your latest commit.
+   
+   e.g., on the appropriate package line:
+   > "net.gree.unity-webview": "https://github.com/SlateScience/unity-webview.git?path=/dist/package#ab84fb0517eb7b3a113465a23ed28fdb5ce92446",
```
please emphasis that  "Packages/packages-lock.json" should change and be pushed

lavikabel commented on plugins/Android/install.sh:
```diff
@@ -1,51 +1,128 @@
 # options
+TARGET="webview"
 MODE="Release"
 SCRIPTING_BACKEND="mono"
```
`SCRIPTING_BACKEND` not in use anymore you can remove

lavikabel commented on plugins/Android/install.sh:
```diff
@@ -1,51 +1,128 @@
+#*)
+#    rm -f ${TARGET}/src/main/java/net/gree/unitywebview/CUnityPlayer*.java
+#    ;;
+#esac
```
please add `END MATIFIC SPECIFIC`

lavikabel reviewed (changes requested):
Great work!!1
Not an easy task, you passed every obstacle thrown at you.

jonobarel-slatescience committed:
Restoring .gitignore file, since deleting it did not solve the problem; tmpl files could still not be committed directly from the build/Packager/Assets/Plugins/Android directory.

This reverts commit f4d43005

jonobarel-slatescience committed:
update readme file.

jonobarel-slatescience committed:
Merge remote-tracking branch 'origin/APPL-8689-merge-updates-from-base-repo' into APPL-8689-merge-updates-from-base-repo

# Conflicts:
#	build/.gitignore

jonobarel-slatescience committed:
restore file storage instructions. APPL-8689

jonobarel-slatescience committed:
update isntructions to include packages-lock. APPL-8689

jonobarel-slatescience committed:
fixes in install.sh
APPL-8689
